docs/PythonAPI.md

```diff
@@ -0,0 +1,867 @@
+        """Checks if two DeviceIdentifier objects are equal."""
+        
+    def __ge__(self, other: object) -> bool:
+        """Checks if this DeviceIdentifier is greater than or equal to another."""
+        
+    def __gt__(self, other: object) -> bool:
+        """Checks if this DeviceIdentifier is greater than another."""
+        
+    def __le__(self, other: object) -> bool:
+        """Checks if this DeviceIdentifier is less than or equal to another."""
+        
+    def __lt__(self, other: object) -> bool:
+        """Checks if this DeviceIdentifier is less than another."""
+        
+    def __ne__(self, other: object) -> bool:
+        """Checks if two DeviceIdentifier objects are not equal."""
+```
+
+- `id`: A tuple containing two integers that serve as the device ID.
+
+- `kind`: An instance of the `DeviceKind` class that represents the type or kind of the device.
+
+- `name`: A string representing the name or label of the device.
+
+- The `dict` method creates a dictionary of a `DeviceIdentifier` object's attributes.
+
+## Class `DeviceKind`
+
+The `DeviceKind` class represents properties for supported devices, such as a camera or video source, in a given system.
+
+```python
+class DeviceKind:
+    Camera: ClassVar[DeviceKind] = DeviceKind.Camera
+    NONE: ClassVar[DeviceKind] = DeviceKind.NONE
+    Signals: ClassVar[DeviceKind] = DeviceKind.Signals
+    StageAxis: ClassVar[DeviceKind] = DeviceKind.StageAxis
+    Storage: ClassVar[DeviceKind] = DeviceKind.Storage
+
+    def __init__(self, *args: None, **kwargs: Any) -> None:
+        """Initializes the DeviceKind class."""
+
+    def __eq__(self, other: object) -> bool:
+        """Checks if two DeviceKind objects are equal."""
+        
+    def __ge__(self, other: object) -> bool:
+        """Checks if this DeviceKind is greater than or equal to another."""
+        
+    def __gt__(self, other: object) -> bool:
+        """Checks if this DeviceKind is greater than another."""
+        
+    def __int__(self) -> int:
+        """Converts the DeviceKind to an integer."""
+        
+    def __le__(self, other: object) -> bool:
+        """Checks if this DeviceKind is less than or equal to another."""
+        
+    def __lt__(self, other: object) -> bool:
+        """Checks if this DeviceKind is less than another."""
+        
+    def __ne__(self, other: object) -> bool:
+        """Checks if two DeviceKind objects are not equal."""
+```
+
+- `Camera`: Class variable of `DeviceKind` that defines the cameras supported by the system. 
+
+- `NONE`: Class variable of `DeviceKind` that is set to None if no device of the specified kind is available.
+
+- `Signals`: Class variable of `DeviceKind` that defines the signals supported by the system.
+
+- `StageAxis`: Class variable of `DeviceKind` that defines the stage axes supported by the system.
+
+- `Storage`: Class variable of `DeviceKind` that defines the storage supported by the system.
```

Comment:
While technically class variables, these are actually enum values.

Reply:
Noted. I changed the language to "Enum-type class variable"

Reply:
Class variables are defined in this file as:
```
Attribute: ClassVariable[Class] = Class.Attribute
```
Are all those class variables in the api of enum-type?